### DISCUSSION TITLE
What is the correct way of Clicking/Invoking a Hyperlink?

### DISCUSSION THREAD

**Discussion by CT4nk3r** [FlaUI/FlaUI#574] [Q&A]
Hi! I have a hyperlink on the UI that I would like to click/invoke.

There is a text over it (in inspect, which seems to be the standard way of doing this), which results in my search not clicking the hyperlink itself (it tries to click the very end if it, which is just text, not a link). I have tried to overcome it and the only way was to calculate the middle of the button and click it there. 

I have been search around a lot and found out that instead of doing mouse buttons I can just Invoke these buttons and hyperlinks. 
In inspect.exe I can even invoke the action of this hyperlink by doing Actions -> Invoke.Invoke. How can I implement this more stable and reliable method to my automation? 

I can find the hyperlink or the text part of it with FindFirstDescendant, but when I try to invoke (there is no .AsHyperlink() method by my knowledge) it errors saying there is not .Invoke for this AutomationElement. I tried converting it into a button, which results in an error.

I don't want to show exact screenshots, because it is a proprietary software but I have found this issue in other programs, so I can show similar use-cases.

**Comment by Anton-V-K** [comment]
Doesn't `AutomationElement.Click()` work for you?
This method works fine for hypelinks in my case.

**Comment by dmcgill50** [comment]
Sounds like it’s button that needs to be invoked. Example code for this is
in the readme for FlaUI.
